player/js/utils/shapes/OffsetPathModifier.js

```diff
@@ -0,0 +1,290 @@
+import {
```

Comment:
This is amazing! I've ran some tests and many of them work great.
There are some cases that break, for example with triangles, or some combinations with other shape modifiers, but it's definitely a great first version. I'd feel comfortable pushing this one as it is and keep improving it when there's feedback.

Reply:
yeah if the path is offset inward too much on sharp corners and the joints overlap (rather than the offset segments) it breaks. My first thought was to check for intersections between line joints and when there is one to remove the segments between them but for some reason I haven't been able to implement it properly